### PR TITLE
Move test packages from `dependencies` to `dev_dependencies`

### PR DESCRIPTION
**Pull request by nt4f04uNd**
Part of https://github.com/flutter/flutter/issues/86894, except for video_player_platform_interface, see https://github.com/flutter/flutter/issues/86894#issuecomment-896853590

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [ ] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by Hixie** [comment]
test-exempt: build configuration refactor with no semantic change

File: packages/in_app_purchase/in_app_purchase_android/pubspec.yaml [hunk 1]
```diff
@@ -22,10 +22,10 @@ dependencies:
   in_app_purchase_platform_interface: ^1.1.0
   json_annotation: ^4.0.1
   meta: ^1.3.0
-  test: ^1.16.0
 
 dev_dependencies:
   build_runner: ^1.11.1
   flutter_test:
     sdk: flutter
+  test: ^1.16.0
```
Comment (stuartmorgan) [inline packages/in_app_purchase/in_app_purchase_android/pubspec.yaml:30]: Please put these in alphabetical order (same for the other files).

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: packages/in_app_purchase/in_app_purchase_android/pubspec.yaml (hunk 1 above)
Comment (nt4f04uNd) [reply packages/in_app_purchase/in_app_purchase_android/pubspec.yaml]: done

**Review by stuartmorgan — APPROVED** [review]
LGTM, thanks!

**Pull request merged**